src/writer/settings.ts

```diff
@@ -73,6 +74,12 @@ export async function verify(body): Promise<any> {
     return Promise.reject(e);
   }
 
+  try {
+    if (ensNormalize(msg.space) !== msg.space.toLowerCase()) throw new Error('');
+  } catch (e) {
+    return Promise.reject('Invalid space id');
+  }
```

Comment:
I think better to do it at ingetor level, else we may have to do this on all actions?